### PR TITLE
Upload build failure log to Github for external contributors access

### PR DESCRIPTION
**Pull request by zhanga5**
~Please do not merge this PR as it still need more testings.~

The log will be uploaded as one of artifacts on github and can be reached via link of blossom-ci Details in case of build failed

### CODE REVIEW

**Comment by zhanga5** [comment]
build

**Comment by zhanga5** [comment]
build

**Comment by zhanga5** [comment]
build

**Comment by tgravescs** [comment]
could you please add more description.  Where exactly do these get uploaded to?  Ie how would user see them.

**Comment by zhanga5** [comment]
> could you please add more description. Where exactly do these get uploaded to? Ie how would user see them.

thanks for your comments. The log will be uploaded as one of artifacts on github and can be reached via link of blossom-ci `Details` in case of build failed

**Comment by tgravescs** [comment]
> > could you please add more description. Where exactly do these get uploaded to? Ie how would user see them.
> 
> thanks for your comments. The log will be uploaded as one of artifacts on github and can be reached via link of blossom-ci `Details` in case of build failed

Note, I meant put it in the PR description at the top.  I updated for you.

File: jenkins/Jenkinsfile-blossom.premerge [hunk 1]
```diff
@@ -191,6 +191,12 @@ pipeline {
                 if (currentBuild.currentResult == "SUCCESS") {
                     githubHelper.updateCommitStatus("$BUILD_URL", "Success", GitHubCommitState.SUCCESS)
                 } else {
+                    // upload log only in case of build failure
+                    def guardWords = ["gitlab.*?\\.com", "urm.*?\\.com"]
+                    def logPattern = "### BEGIN OF TEST LOG ###.*### END OF TEST LOG ###"
+
+                    githubHelper.uploadPartialLogs(this, env.JOB_NAME, env.BUILD_NUMBER, null, guardWords, logPattern)
```
Comment (tgravescs) [inline jenkins/Jenkinsfile-blossom.premerge:198]: where can I find the docs for uploadPartialLogs?

Comment (zhanga5) [reply jenkins/Jenkinsfile-blossom.premerge:198]: replied the link to you via email

**Review by tgravescs — APPROVED** [review]

**Comment by zhanga5** [comment]
build

**Comment by pxLi** [comment]
build

**Pull request merged**